dart/lib/flat_buffers.dart

```diff
@@ -335,8 +339,8 @@ class Builder {
   ///
   /// Most clients should prefer calling [finish].
   Uint8List lowFinish() {
-    int alignedTail = _tail + ((-_tail) % _maxAlign);
-    return _buf.buffer.asUint8List(_buf.lengthInBytes - alignedTail);
+    size = _tail + ((-_tail) % _maxAlign);
```

Comment:
Instead, make a `size` method that returns `_tail + ((-_tail) % _maxAlign)` ? That can then be called from these two locations, and also externally, and also has the benefit of not having an extra instance variable that can be incorrect.

Reply:
That means that every call to size (and there are many) will have to do the computation.

Reply:
There are already 2 locations in the code where `_tail + ((-_tail) % _maxAlign)` is computed.. replacing those with a function call will not make things worse other than the function call itself. Other users of this function don't exist yet, so are also not making things slower. Note that `size` is being introduced in this PR.

Reply:
Ahhh sorry about that, I'm thinking of somewhere else that similar logic exists I think - crossing PRs :)

Reply:
If it's exposed publicly though it will involve a closure and computation every time it's invoked by public users.

Reply:
well, doesn't sound like its needed frequently, so it seems cleaner to me. But if all of this is expensive in Dart, I am fine with the extra variable too.

Reply:
Yeah, well that's how I've done it originally (having a getter that does the calculation) in the other PR before changing after @dnfield's comments and moving here :shrug:

Reply:
Ahh so I'm not mixing it up. This getter will be called in multiple places during building. It's better to not require the re-computation on each call.

Reply:
The current version should be to your liking :) 

However, I've checked again and it's actually not "called in multiple places during building" - just once at the end, when you finish() a complete buffer

Reply:
Oh goodness.  Haha ok then, let's just move the logic to the closure then. Sorry for the confusion.

Reply:
No problem, rolled back. 

I bet you'll want to use a variable in finish() :D